feat: Transifex - Contentful Integration app

foteinigk opened the pull request:
## Purpose

We are Transifex and we would like to have our app in your Marketplace. With the Transifex app for Contentful, you can easily sync content for localization between Transifex and Contentful and leverage all the localization management tools Transifex offers. Experience a streamlined approach to managing your multilingual content with Transifex, enhancing your global content delivery.

**Features**
- Localize specific content blocks with ease by selecting them for translation and sending them directly to Transifex.
- Notify your localization team automatically for any source content updates.
- Set the required translation status (translated, reviewed, proofread) for automatic syncing of localized content or any updates thereafter.
- Help translators pick up where they left off, by sending existing translations to Transifex.
- Supported field types: Text, Rich Text, Reference. This applies for all your Entries and Assets.


## Approach
The Transifex application includes:

- **Configuration screen:**  
The configuration necessary for the application to work properly. To use transifex app, a Transifex account and a project already created in Transifex is required.  
![Screenshot 2023-12-08 at 2 08 21 PM](https://github.com/contentful/marketplace-partner-apps/assets/112862979/652f6e96-a662-4786-a69b-58de7fcefac6)


After logging into Transifex, you can configure content synchronisation options and content types. Additionally, the connected Transifex project will be displayed.

![Screenshot 2023-12-08 at 2 52 40 PM](https://github.com/contentful/marketplace-partner-apps/assets/112862979/559e6c30-8d86-4b17-b880-ae375c96aed4)


- **Sidebar**
Once you have installed the Transifex app in your Contentful space, you can start using the Transifex sidebar in Contentful’s Entry editor.  You can import an entry  and the respective resource will be created into Transifex. 
You can see the translation progress in all target languages for this specific Entry:

![Screenshot 2023-12-08 at 2 53 56 PM](https://github.com/contentful/marketplace-partner-apps/assets/112862979/cd15cad8-4347-43f7-b37e-3f7f846bc23c)

## Testing steps
- Create a Transifex account/project if you don't have already
- Install app through this link:
   https://app.contentful.com/deeplink?link=apps&id=1EJ1VgJTJ6C3OIiXie2l2F (production app) 
   https://app.contentful.com/deeplink?link=apps&id=6C1iuvnKYPEDjbMsoBFdFb (staging app)

## Deploy
- Staging
`npm install`
`npm run build:dev`
`npm run upload`

- Staging
`npm install`
`npm run build:staging`
`npm run upload`
`npm run deploy:staging`

- Production
`npm install`
`npm run build`
`npm run upload`
`npm run deploy`

lilbitner commented on apps/transifex/README.md:
```diff
@@ -0,0 +1,44 @@
+## Deployment
+
+Local
+To upload the bundle of local app to Contnetful
```
just a few small typos here with Contentful :) (and below)

lilbitner reviewed (commented):
Hey there! The app looks great, just added a small comment to remove the deploy scripts for now. These will be manually updated by us to point to a newly create app definition 😃

lilbitner commented on apps/transifex/.env.staging:
```diff
@@ -0,0 +1,3 @@
+VITE_BACKEND_HOST=https://contentfulapp.stg.transifex.net
+VITE_CONTENTFUL_APP_HOST=https://contentfulapp.stg.transifex.net
+VITE_TXC_HOST=https://app.transifex.com
```
At this point we don't have an official staging environment to host this, so I am not sure it is necessary to have staging references in this code at this time. But it is up to you. We plan on implementing an official MPA Staging environment where we will can then create a new app definition ID for this to deploy to.

foteinigk commented:
> Contnetful

Thank you very much! I have made the corrections, I just have one question. We [verify requests ](https://www.contentful.com/developers/docs/extensibility/app-framework/request-verification/) so we use the signing secret and app definition ID in the backend code. Will we have access to the new application definition so we can update the relevant variables accordingly in our code?

lilbitner commented:
> Thank you very much! I have made the corrections, I just have one question. We [verify requests ](https://www.contentful.com/developers/docs/extensibility/app-framework/request-verification/), so we use the signing secret and app definition ID in the backend code. Will we have access to the new application definition so we can update the relevant variables accordingly in our code?

Thank you for making those changes! Unfortunately at this time you will not have access to the app definition directly, but I will happily provide you with the secure key pair via a short-term 1Password link, as well as a the app definition id. I will prepare this and share with you before deploying the app!

foteinigk commented:
> > Thank you very much! I have made the corrections, I just have one question. We [verify requests ](https://www.contentful.com/developers/docs/extensibility/app-framework/request-verification/), so we use the signing secret and app definition ID in the backend code. Will we have access to the new application definition so we can update the relevant variables accordingly in our code?
> 
> Thank you for making those changes! Unfortunately at this time you will not have access to the app definition directly, but I will happily provide you with the secure key pair via a short-term 1Password link, as well as a the app definition id. I will prepare this and share with you before deploying the app!

Thank you 🙏.  Also, will the selected registered events be copied to the new app definition?  Currently we use only the "App Installation/Delete" with the following url "https://contentful.apps.transifex.com/api/subscribed-events/app-installations"

lilbitner commented:
> > > Thank you very much! I have made the corrections, I just have one question. We [verify requests ](https://www.contentful.com/developers/docs/extensibility/app-framework/request-verification/), so we use the signing secret and app definition ID in the backend code. Will we have access to the new application definition so we can update the relevant variables accordingly in our code?
> > 
> > 
> > Thank you for making those changes! Unfortunately at this time you will not have access to the app definition directly, but I will happily provide you with the secure key pair via a short-term 1Password link, as well as a the app definition id. I will prepare this and share with you before deploying the app!
> 
> Thank you 🙏. Also, will the selected registered events be copied to the new app definition? Currently we use only the "App Installation/Delete" with the following url "https://contentful.apps.transifex.com/api/subscribed-events/app-installations"

Ah yes thank you for clarifying this. Please verify that the below screenshot is correct: 

![Screenshot 2024-01-04 at 9 11 25 AM](https://github.com/contentful/marketplace-partner-apps/assets/58186851/b064ce3d-400b-453d-9add-244e57518077)

Additional configurations include that the app exists on the Configuration page and the Sidebar locations. Are there any other configurations specific to this app that I should add, other than the key pair?

foteinigk commented:
> > > > Thank you very much! I have made the corrections, I just have one question. We [verify requests ](https://www.contentful.com/developers/docs/extensibility/app-framework/request-verification/), so we use the signing secret and app definition ID in the backend code. Will we have access to the new application definition so we can update the relevant variables accordingly in our code?
> > > 
> > > 
> > > Thank you for making those changes! Unfortunately at this time you will not have access to the app definition directly, but I will happily provide you with the secure key pair via a short-term 1Password link, as well as a the app definition id. I will prepare this and share with you before deploying the app!
> > 
> > 
> > Thank you 🙏. Also, will the selected registered events be copied to the new app definition? Currently we use only the "App Installation/Delete" with the following url "https://contentful.apps.transifex.com/api/subscribed-events/app-installations"
> 
> Ah yes thank you for clarifying this. Please verify that the below screenshot is correct:
> 
> ![Screenshot 2024-01-04 at 9 11 25 AM](https://private-user-images.githubusercontent.com/58186851/294234999-b064ce3d-400b-453d-9add-244e57518077.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQzODUyNzcsIm5iZiI6MTcwNDM4NDk3NywicGF0aCI6Ii81ODE4Njg1MS8yOTQyMzQ5OTktYjA2NGNlM2QtNDAwYi00NTNkLTlhZGQtMjQ0ZTU3NTE4MDc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDE2MTYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmMWJhNGJhMGUxZDVjNGY0ODY0OTE4MzRmZTg4MTUyZTkwMzFmOTVmMTJmNmY1ZTZmMGQ5NjRkMTQ2MWEyZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OsebgSn-OWz22w0VfmI-rjpyri37-XEVONRPPCiZbQE)
> 
> Additional configurations include that the app exists on the Configuration page and the Sidebar locations. Are there any other configurations specific to this app that I should add, other than the key pair?
> 
> And do you mind providing a good email to share the secure key pair link?

Yeh sure,  screenshot looks fine , we use only the delete action for app installation event and the url is the right one.  We have also enabled the option verify requests and we use the signing secret at our backend code. So we would need the app definition ID and the signing secret. This is my email foteini@transifex.com . Thanks a lot!!

lilbitner commented:
> > > > > Thank you very much! I have made the corrections, I just have one question. We [verify requests ](https://www.contentful.com/developers/docs/extensibility/app-framework/request-verification/), so we use the signing secret and app definition ID in the backend code. Will we have access to the new application definition so we can update the relevant variables accordingly in our code?
> > > > 
> > > > 
> > > > Thank you for making those changes! Unfortunately at this time you will not have access to the app definition directly, but I will happily provide you with the secure key pair via a short-term 1Password link, as well as a the app definition id. I will prepare this and share with you before deploying the app!
> > > 
> > > 
> > > Thank you 🙏. Also, will the selected registered events be copied to the new app definition? Currently we use only the "App Installation/Delete" with the following url "https://contentful.apps.transifex.com/api/subscribed-events/app-installations"
> > 
> > 
> > Ah yes thank you for clarifying this. Please verify that the below screenshot is correct:
> > ![Screenshot 2024-01-04 at 9 11 25 AM](https://private-user-images.githubusercontent.com/58186851/294234999-b064ce3d-400b-453d-9add-244e57518077.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQzODUyNzcsIm5iZiI6MTcwNDM4NDk3NywicGF0aCI6Ii81ODE4Njg1MS8yOTQyMzQ5OTktYjA2NGNlM2QtNDAwYi00NTNkLTlhZGQtMjQ0ZTU3NTE4MDc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDE2MTYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmMWJhNGJhMGUxZDVjNGY0ODY0OTE4MzRmZTg4MTUyZTkwMzFmOTVmMTJmNmY1ZTZmMGQ5NjRkMTQ2MWEyZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OsebgSn-OWz22w0VfmI-rjpyri37-XEVONRPPCiZbQE)
> > Additional configurations include that the app exists on the Configuration page and the Sidebar locations. Are there any other configurations specific to this app that I should add, other than the key pair?
> > And do you mind providing a good email to share the secure key pair link?
> 
> Yeh sure, screenshot looks fine , we use only the delete action for app installation event and the url is the right one. We have also enabled the option verify requests and we use the signing secret at our backend code. So we would need the app definition ID and the signing secret. This is my email [foteini@transifex.com](mailto:foteini@transifex.com) . Thanks a lot!!

Awesome! I went ahead and sent an email! I also realize that it looks like I don't have permission to push to the forked branch here in order to update the deploy script. Do you mind adding the following script within your package.json, where the previous deploy scripts were: 

    "deploy": "contentful-app-scripts upload --ci --bundle-dir ./build --organization-id ${DEFINITIONS_ORG_ID} --definition-id b7SVv0k1EYbQRsusviQDi --token ${CONTENTFUL_CMA_TOKEN}"

Thank you in advance!
Once this is adjusted, and once your backend app has been adjusted with the correct secret and definition id, we can go ahead and merge this PR. The actual release will occur in a few follow-up PRs after.

foteinigk commented:
> > > > > > Thank you very much! I have made the corrections, I just have one question. We [verify requests ](https://www.contentful.com/developers/docs/extensibility/app-framework/request-verification/), so we use the signing secret and app definition ID in the backend code. Will we have access to the new application definition so we can update the relevant variables accordingly in our code?
> > > > > 
> > > > > 
> > > > > Thank you for making those changes! Unfortunately at this time you will not have access to the app definition directly, but I will happily provide you with the secure key pair via a short-term 1Password link, as well as a the app definition id. I will prepare this and share with you before deploying the app!
> > > > 
> > > > 
> > > > Thank you 🙏. Also, will the selected registered events be copied to the new app definition? Currently we use only the "App Installation/Delete" with the following url "https://contentful.apps.transifex.com/api/subscribed-events/app-installations"
> > > 
> > > 
> > > Ah yes thank you for clarifying this. Please verify that the below screenshot is correct:
> > > ![Screenshot 2024-01-04 at 9 11 25 AM](https://private-user-images.githubusercontent.com/58186851/294234999-b064ce3d-400b-453d-9add-244e57518077.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQzODUyNzcsIm5iZiI6MTcwNDM4NDk3NywicGF0aCI6Ii81ODE4Njg1MS8yOTQyMzQ5OTktYjA2NGNlM2QtNDAwYi00NTNkLTlhZGQtMjQ0ZTU3NTE4MDc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDE2MTYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmMWJhNGJhMGUxZDVjNGY0ODY0OTE4MzRmZTg4MTUyZTkwMzFmOTVmMTJmNmY1ZTZmMGQ5NjRkMTQ2MWEyZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OsebgSn-OWz22w0VfmI-rjpyri37-XEVONRPPCiZbQE)
> > > Additional configurations include that the app exists on the Configuration page and the Sidebar locations. Are there any other configurations specific to this app that I should add, other than the key pair?
> > > And do you mind providing a good email to share the secure key pair link?
> > 
> > 
> > Yeh sure, screenshot looks fine , we use only the delete action for app installation event and the url is the right one. We have also enabled the option verify requests and we use the signing secret at our backend code. So we would need the app definition ID and the signing secret. This is my email [foteini@transifex.com](mailto:foteini@transifex.com) . Thanks a lot!!
> 
> Awesome! I went ahead and sent an email! I also realize that it looks like I don't have permission to push to the forked branch here in order to update the deploy script. Do you mind adding the following script within your package.json, where the previous deploy scripts were:
> 
> ```
> "deploy": "contentful-app-scripts upload --ci --bundle-dir ./build --organization-id ${DEFINITIONS_ORG_ID} --definition-id b7SVv0k1EYbQRsusviQDi --token ${CONTENTFUL_CMA_TOKEN}"
> ```
> 
> Thank you in advance! Once this is adjusted, and once your backend code has been adjusted with the correct secret and definition id, we can go ahead and merge this PR. The actual release will occur in a few follow-up PRs after.

Thank you!!🙏 We have the new secrets and we have updated the deploy script. Before we release the app to the marketplace, would we have a chance to test  it once again to make sure everything works as expected with the new configuration?

foteinigk commented:
> > > > > > > Thank you very much! I have made the corrections, I just have one question. We [verify requests ](https://www.contentful.com/developers/docs/extensibility/app-framework/request-verification/), so we use the signing secret and app definition ID in the backend code. Will we have access to the new application definition so we can update the relevant variables accordingly in our code?
> > > > > > 
> > > > > > 
> > > > > > Thank you for making those changes! Unfortunately at this time you will not have access to the app definition directly, but I will happily provide you with the secure key pair via a short-term 1Password link, as well as a the app definition id. I will prepare this and share with you before deploying the app!
> > > > > 
> > > > > 
> > > > > Thank you 🙏. Also, will the selected registered events be copied to the new app definition? Currently we use only the "App Installation/Delete" with the following url "https://contentful.apps.transifex.com/api/subscribed-events/app-installations"
> > > > 
> > > > 
> > > > Ah yes thank you for clarifying this. Please verify that the below screenshot is correct:
> > > > ![Screenshot 2024-01-04 at 9 11 25 AM](https://private-user-images.githubusercontent.com/58186851/294234999-b064ce3d-400b-453d-9add-244e57518077.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQzODUyNzcsIm5iZiI6MTcwNDM4NDk3NywicGF0aCI6Ii81ODE4Njg1MS8yOTQyMzQ5OTktYjA2NGNlM2QtNDAwYi00NTNkLTlhZGQtMjQ0ZTU3NTE4MDc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDE2MTYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmMWJhNGJhMGUxZDVjNGY0ODY0OTE4MzRmZTg4MTUyZTkwMzFmOTVmMTJmNmY1ZTZmMGQ5NjRkMTQ2MWEyZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OsebgSn-OWz22w0VfmI-rjpyri37-XEVONRPPCiZbQE)
> > > > Additional configurations include that the app exists on the Configuration page and the Sidebar locations. Are there any other configurations specific to this app that I should add, other than the key pair?
> > > > And do you mind providing a good email to share the secure key pair link?
> > > 
> > > 
> > > Yeh sure, screenshot looks fine , we use only the delete action for app installation event and the url is the right one. We have also enabled the option verify requests and we use the signing secret at our backend code. So we would need the app definition ID and the signing secret. This is my email [foteini@transifex.com](mailto:foteini@transifex.com) . Thanks a lot!!
> > 
> > 
> > Awesome! I went ahead and sent an email! I also realize that it looks like I don't have permission to push to the forked branch here in order to update the deploy script. Do you mind adding the following script within your package.json, where the previous deploy scripts were:
> > ```
> > "deploy": "contentful-app-scripts upload --ci --bundle-dir ./build --organization-id ${DEFINITIONS_ORG_ID} --definition-id b7SVv0k1EYbQRsusviQDi --token ${CONTENTFUL_CMA_TOKEN}"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Thank you in advance! Once this is adjusted, and once your backend code has been adjusted with the correct secret and definition id, we can go ahead and merge this PR. The actual release will occur in a few follow-up PRs after.
> 
> Thank you!!🙏 We have the new secrets and we have updated the deploy script. Before we release the app to the marketplace, would we have a chance to test it once again to make sure everything works as expected with the new configuration?

For your information, we have updated the new secrets on our backend code🙂

matthew-gordon commented:
> > > > > > > > Thank you very much! I have made the corrections, I just have one question. We [verify requests ](https://www.contentful.com/developers/docs/extensibility/app-framework/request-verification/), so we use the signing secret and app definition ID in the backend code. Will we have access to the new application definition so we can update the relevant variables accordingly in our code?
> > > > > > > 
> > > > > > > 
> > > > > > > Thank you for making those changes! Unfortunately at this time you will not have access to the app definition directly, but I will happily provide you with the secure key pair via a short-term 1Password link, as well as a the app definition id. I will prepare this and share with you before deploying the app!
> > > > > > 
> > > > > > 
> > > > > > Thank you 🙏. Also, will the selected registered events be copied to the new app definition? Currently we use only the "App Installation/Delete" with the following url "https://contentful.apps.transifex.com/api/subscribed-events/app-installations"
> > > > > 
> > > > > 
> > > > > Ah yes thank you for clarifying this. Please verify that the below screenshot is correct:
> > > > > ![Screenshot 2024-01-04 at 9 11 25 AM](https://private-user-images.githubusercontent.com/58186851/294234999-b064ce3d-400b-453d-9add-244e57518077.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQzODUyNzcsIm5iZiI6MTcwNDM4NDk3NywicGF0aCI6Ii81ODE4Njg1MS8yOTQyMzQ5OTktYjA2NGNlM2QtNDAwYi00NTNkLTlhZGQtMjQ0ZTU3NTE4MDc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDE2MTYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmMWJhNGJhMGUxZDVjNGY0ODY0OTE4MzRmZTg4MTUyZTkwMzFmOTVmMTJmNmY1ZTZmMGQ5NjRkMTQ2MWEyZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OsebgSn-OWz22w0VfmI-rjpyri37-XEVONRPPCiZbQE)
> > > > > Additional configurations include that the app exists on the Configuration page and the Sidebar locations. Are there any other configurations specific to this app that I should add, other than the key pair?
> > > > > And do you mind providing a good email to share the secure key pair link?
> > > > 
> > > > 
> > > > Yeh sure, screenshot looks fine , we use only the delete action for app installation event and the url is the right one. We have also enabled the option verify requests and we use the signing secret at our backend code. So we would need the app definition ID and the signing secret. This is my email [foteini@transifex.com](mailto:foteini@transifex.com) . Thanks a lot!!
> > > 
> > > 
> > > Awesome! I went ahead and sent an email! I also realize that it looks like I don't have permission to push to the forked branch here in order to update the deploy script. Do you mind adding the following script within your package.json, where the previous deploy scripts were:
> > > ```
> > > "deploy": "contentful-app-scripts upload --ci --bundle-dir ./build --organization-id ${DEFINITIONS_ORG_ID} --definition-id b7SVv0k1EYbQRsusviQDi --token ${CONTENTFUL_CMA_TOKEN}"
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Thank you in advance! Once this is adjusted, and once your backend code has been adjusted with the correct secret and definition id, we can go ahead and merge this PR. The actual release will occur in a few follow-up PRs after.
> > 
> > 
> > Thank you!!🙏 We have the new secrets and we have updated the deploy script. Before we release the app to the marketplace, would we have a chance to test it once again to make sure everything works as expected with the new configuration?
> 
> For your information, we have updated the new secrets on our backend code🙂

Thank you kindly, we will look to get this merged in ASAP, I will connect with @lilbitner to see if its possible to test again before final release!
